### ISSUE TITLE
Issues with seeing variables in Debug

### ISSUE THREAD

**Issue by elichai** [intellij-rust/intellij-rust#3235]
## Environment

* **Intellij-Rust plugin version:** v0.2.0.2113-183
* **Rust toolchain version:** 1.31.1
* **IDE name and version:** CLion 2018.3.2
* **Operating system:** Ubuntu 18.04, Linux 4.15.18-041518-generic

## Problem description
When I debug code a lot of time when it gets into a function I can't see any of the variables, I see only `Duplicate variable object name`

## Steps to reproduce
Just try to debug any big project.

**Comment by CallistoM** [comment]
Same problem, no variables are shown: 

### Environment
- Intellij-Rust plugin version: v0.2.0.2113-183
- Rust toolchain version: 1.31.1
- IDE name and version: CLion 2018.3.2
- Operating system: Mac OS 10.14.2 (18C54)

**Comment by ortem** [comment]
1. Could you please provide some source code snippet?
2. Did you try to debug such programs with command-line gdb/lldb? I should know whether this problem relates to IDE or to the debugger itself.
3. Are the variables not shown only on some steps, or they aren't shown during the whole function body?

**Comment by CallistoM** [comment]
1. See: <img width="1008" alt="schermafbeelding 2019-01-16 om 19 48 15" src="https://user-images.githubusercontent.com/11871820/51271387-b5c0ab80-19c7-11e9-9b57-d39c9865f82f.png">
2. No, not yet.
3. During the whole body.

**Comment by ortem** [comment]
@CallistoM Thanks for the information! I reproduced the issue, and also found some strange cases (e.g. it works for `i32`, but doesn't work for `f64`).
<img width="446" alt="screenshot 2019-01-17 at 12 22 44" src="https://user-images.githubusercontent.com/4854600/51308950-5e1c5180-1a54-11e9-95bb-e1bc60e4eeca.png">
<img width="305" alt="screenshot 2019-01-17 at 12 23 09" src="https://user-images.githubusercontent.com/4854600/51308951-5e1c5180-1a54-11e9-8868-cc3ed7ab32e4.png">

2. I tried command-line lldb and it works fine, so the problem is on the IDE side.

3. In my case, when I set a breakpoint on the first function body line and repeat `Step over` (F8), there are no showed variables only on some special points, but not on the whole function body.

In summary, there is a bug and I'll try to fix it, thanks for the report! As a temporary solution, you can open `LLDB` tab (to the right of `Variables` tab inside `Debug` tool window) and enter `frame variable` to the lldb command line. Then you will see all the variables debugger provides at this point. Also, to watch a particular variable in `Variables` tab (for example, if you want to look inside a vector's elements and `lldb` console output is unhandy), you can use `New Watch` ("plus" button inside the `Variables` tab).

**Comment by CallistoM** [comment]
@ortem Thanks for the tip, i will use the lldb console for now!

**Comment by amakeev** [comment]
@CallistoM Also a workaround might help: Uncheck `Debugger | Data Views | C++ | Hide out-of-scope variables`.

**Comment by apouche** [comment]
I'm having the exact same issue using CLion, apparently it was solved a long time ago. Maybe I'm missing something ?

![Screen Shot 2020-12-18 at 12 25 57 AM](https://user-images.githubusercontent.com/528492/102555725-9e186300-40c7-11eb-9276-8ad44878491e.png)

**Comment by apouche** [comment]
nevermind, I put the breakpoint too late in the hidden return statement and I guess at this point the stack associated with this function was already deallocated. Sorry for the noise
